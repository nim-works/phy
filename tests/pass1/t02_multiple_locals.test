discard """
  description: "Test a single continuation with more than one blob local"
"""
(TypeDefs
  (Blob 16)
  (ProcTy (Void))
)
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 1)
    (Locals
      (Type 0)
      (Type 0)
    )
    (Continuations
      (Continuation (Params) (Locals (Local 0) (Local 1))
        (Continue 1)
      )
      (Continuation (Params))
    )
  )
)