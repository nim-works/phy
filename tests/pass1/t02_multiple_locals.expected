Module
  TypeDefs
    UInt
      Immediate 8
    ProcTy
      Void
  GlobalDefs
  ProcDefs
    ProcDef
      Type 1
      Locals
        Type 0
        Type 0
      Continuations
        Continuation
          Params
          Immediate 32
          Asgn
            Local 0
            Addr
              IntVal 0
          Asgn
            Local 1
            Addr
              IntVal 16
          Continue
            Immediate 1
        Continuation
          Params
